### PR TITLE
Event Sourcing

### PR DESCRIPTION
**Pull request by tsb99x**
Move to event sourcing approach to improve the tasks storage. Think about moving to the IndexedDB in the future. Use Vuex-like state, getters and actions in components.

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -41,70 +41,23 @@
...
-            doneTasks: function() {
-                return this.tasks.filter(task => task.isDone)
-            }
-        },
-        mounted() {
-            this.loadTasks()
+            ...getters
         },
         methods: {
+            ...actions,
```
Comment (GKsenon) [inline src/App.vue:57]: It's not obvious which actions and getters are used by the component.

File: src/state.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        }
+    },
+    activateTask: {
+        v1(event) {
+            state.tasks[event.entityId].isDone = false
+        }
+    }
+}
+
+let events = loadJson(EVENTS, [])
```
Comment (GKsenon) [inline src/state.js:56]: An explicit initialization through a function call would be easier to understand.

**Review by GKsenon — CHANGES_REQUESTED** [review]

File: src/App.vue (hunk 1 above)
Comment (tsb99x) [reply src/App.vue]: Yes, refactored to call actions explicitly through the component's methods.

File: src/state.js (hunk 2 above)
Comment (tsb99x) [reply src/state.js]: Yup, wrapped the initialization into an appropriate function. A call to this function is done at the `main.js`.

**Review by GKsenon — APPROVED** [review]

**Pull request merged**